### DISCUSSION TITLE
Add an option to display tokens/s in real-time for both side-by-side and single model inference.

### DISCUSSION THREAD

**Discussion by zytoh0** [open-webui/open-webui#5451] [Ideas]
**Is your feature request related to a problem? Please describe.**
Open WebUI’s comparison feature is useful for evaluating output quality, but it lacks the ability to show tokens per second (tokens/s), which is important for comparing model speed. This is missing in both side-by-side and single model inference. UIs like Jan.ai already include this feature, making it easier to assess model performance. (See image below). Visit https://www.youtube.com/watch?v=QpMQgJL4AZA and scroll to 02:48 to see it display token/s in real-time.
![image](https://github.com/user-attachments/assets/5a397aa8-ff7a-4a17-9665-14387b87137b)

**Describe the solution you'd like**
Add a setting that displays tokens/s in real-time for both side-by-side comparisons and single model inferences. This would allow users to track model speed alongside output quality.

**Describe alternatives you've considered**
Another option could be to display tokens/s within the response itself, either below or next to the text for every single response but having a setting to toggle this feature on or off would be more flexible for users who may not always need this information.

**Additional context**
Adding this feature would greatly improve the usability of Open WebUI by allowing users to evaluate both model output quality and processing speed. This is especially useful when speed is a critical factor.

**Comment by Wangs121** [comment]
there is a function [time_token_tracker](https://openwebui.com/f/owndev/time_token_tracker) could show response time, request &response token and token speed
![Snipaste_2025-02-22_10-49-14](https://github.com/user-attachments/assets/ee4a8ea8-e9e9-4bca-b8ca-b743a9567857)

**Reply by zytoh0** [reply]
Thanks. They've implemented this request. Closing this as resolved.
